tests/redux/actions/channel_actions.test.js

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import thunk from 'redux-thunk';
+import configureStore from 'redux-mock-store';
+
+import * as Actions from 'actions/channel_actions';
+
+const mockStore = configureStore([thunk]);
+
+jest.mock('mattermost-redux/actions/channels', () => ({
+    fetchMyChannelsAndMembers: (...args) => ({type: 'MOCK_FETCH_CHANNELS_AND_MEMBERS', args}),
+}));
+
+jest.mock('actions/user_actions.jsx', () => ({
+    loadNewDMIfNeeded: jest.fn(),
+    loadNewGMIfNeeded: jest.fn(),
+    loadProfilesForSidebar: jest.fn(),
+}));
+
+describe('Actions.Channel', () => {
+    const initialState = {
+        entities: {
+            channels: {
+                currentChannelId: 'current_channel_id',
+                myMembers: {
+                    current_channel_id: {
+                        channel_id: 'current_channel_id',
+                        user_id: 'current_user_id',
+                        roles: 'channel_role',
+                        mention_count: 1,
+                        msg_count: 9,
+                    },
+                },
+                channels: {
+                    current_channel_id: {
+                        total_msg_count: 10,
+                        team_id: 'team_id',
+                    },
+                },
+            },
+            teams: {
+                currentTeamId: 'team-id',
+                teams: {
+                    team_id: {
+                        id: 'team_id',
+                        name: 'team-1',
+                        displayName: 'Team 1',
+                    },
+                },
+                myMembers: {
+                    'team-id': {roles: 'team_role'},
+                },
+            },
+            users: {
+                currentUserId: 'current_user_id',
+                profiles: {
+                    current_user_id: {roles: 'system_role'},
+                },
+            },
+            roles: {
+                roles: {
+                    system_role: {
+                        permissions: [],
+                    },
+                    team_role: {
+                        permissions: [],
+                    },
+                    channel_role: {
+                        permissions: [],
+                    },
+                },
+            },
+        },
+    };
+
+    test('loadChannelsForCurrentUser', async () => {
+        const testStore = await mockStore(initialState);
+
+        const expectedActions = [{
+            type: 'MOCK_FETCH_CHANNELS_AND_MEMBERS',
+            args: ['team-id'],
+        }];
+
+        await testStore.dispatch(Actions.loadChannelsForCurrentUser());
+        expect(testStore.getActions()).toEqual(expectedActions);
```

Comment:
I also don't see much to test, maybe add whether the `loadProfilesForSidebar` has been called.
```javascript
import {loadProfilesForSidebar} from 'actions/user_actions.jsx';
...
expect(loadProfilesForSidebar).toHaveBeenCalledTimes(1);
```